src/components/ToggleZoomOverlay/ToggleZoomOverlay.js

```diff
@@ -0,0 +1,104 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import core from 'core';
+
+import ToggleElementButton from 'components/ToggleElementButton';
+import { zoomTo } from 'helpers/zoom';
+import selectors from 'selectors';
+import actions from 'actions';
+
+import './ToggleZoomOverlay.scss';
+
+const propTypes = {
+  onClick: PropTypes.func.isRequired,
+  isActive: PropTypes.bool,
+};
+
+const ToggleZoomOverlay = ({
+  onClick,
+  isActive,
+}) => {
+  const [value, setValue] = useState('100');
+
+  useEffect(() => {
+    const onDocumentLoaded = () => setValue(Math.ceil(core.getZoom() * 100).toString());
+    const onZoomUpdated = () => setValue(Math.ceil(core.getZoom() * 100).toString());
+
+    core.addEventListener('documentLoaded', onDocumentLoaded);
+    core.addEventListener('zoomUpdated', onZoomUpdated);
+
+    return () => {
+      core.removeEventListener('documentLoaded', onDocumentLoaded);
+      core.removeEventListener('zoomUpdated', onZoomUpdated);
+    };
+  });
```

Comment:
I don't think we actually need to remove the listeners and then add them back on every rerender. Do you think we can just use `[]` as the second argument?